src/Twig/Node/CSPNode.php

```diff
@@ -22,40 +22,60 @@
  * file that was distributed with this source code.
  */
 
+use Twig\Attribute\YieldReady;
 use Twig\Compiler;
+use Twig\Node\CaptureNode;
 use Twig\Node\Node;
 
+#[YieldReady]
 final class CSPNode extends Node
 {
     private ?string $sha;
     private string $directive;
 
     public function __construct(Node $body, int $lineno, string $tag, string $directive, ?string $sha = null)
     {
+        if (class_exists(CaptureNode::class)) {
+            $body = new CaptureNode($body, $lineno, $tag);
+            $body->setAttribute('raw', true);
+        }
+
         parent::__construct(['body' => $body], [], $lineno, $tag);
         $this->sha = $sha;
         $this->directive = $directive;
     }
 
     public function compile(Compiler $compiler): void
     {
-        $body = $this->getNode('body');
+        if (class_exists(CaptureNode::class)) {
+            $compiler
+                ->addDebugInfo($this)
+                ->indent()
+                ->write("\$content = ")
+                ->subcompile($this->getNode('body'))
+                ->raw("\n")
+                ->outdent()
+            ;
+        } else {
+            $compiler
+                ->addDebugInfo($this)
+                ->indent()
+                ->write("ob_start();\n")
+                ->subcompile($this->getNode('body'))
+                ->outdent()
+                ->write("\$content = ob_get_clean();\n")
+            ;
+        }
 
         if (null !== $this->sha) {
-            $output = "\$this->env->getRuntime('Nelmio\SecurityBundle\Twig\CSPRuntime')->getListener()->addSha('{$this->directive}', '{$this->sha}');\necho ob_get_clean();\n";
+            $compiler->write("\$this->env->getRuntime('Nelmio\SecurityBundle\Twig\CSPRuntime')->getListener()->addSha('{$this->directive}', '{$this->sha}');\n");
         } elseif ('script-src' === $this->directive) {
-            $output = "\$script = ob_get_clean();\n\$this->env->getRuntime('Nelmio\SecurityBundle\Twig\CSPRuntime')->getListener()->addScript(\$script);\necho \$script;\n";
+            $compiler->write("\$this->env->getRuntime('Nelmio\SecurityBundle\Twig\CSPRuntime')->getListener()->addScript(\$content);\n");
         } elseif ('style-src' === $this->directive) {
-            $output = "\$style = ob_get_clean();\n\$this->env->getRuntime('Nelmio\SecurityBundle\Twig\CSPRuntime')->getListener()->addStyle(\$style);\necho \$style;\n";
+            $compiler->write("\$this->env->getRuntime('Nelmio\SecurityBundle\Twig\CSPRuntime')->getListener()->addStyle(\$content);\n");
         } else {
             throw new \InvalidArgumentException(sprintf('Unable to compile for directive "%s"', $this->directive));
         }
-
-        $compiler
-            ->addDebugInfo($this)
-            ->write("ob_start();\n")
-            ->subcompile($body)
-            ->write($output)
-        ;
+        $compiler->write("echo \$content;\n");
```

Comment:
I still get deprecation warnings in the test suite which I think are coming from this line?

  4x: Since twig/twig 3.9: Using "echo" is deprecated, use "yield" instead in "Nelmio\SecurityBundle\Twig\Node\CSPNode", then flag the class with #[YieldReady].
    2x in IntegrationTest::testItWorksDynamically from Nelmio\SecurityBundle\Tests\Twig
    2x in IntegrationTest::testItWorksStatically from Nelmio\SecurityBundle\Tests\Twig

Should this not be a yield?

Reply:
indeed :(